handler-proxy/src/test/java/io/netty/handler/proxy/HttpProxyHandlerTest.java

```diff
@@ -153,6 +176,49 @@ public void testCustomHeaders() throws Exception {
                 true);
     }
 
+    @Test
+    public void testExceptionDuringConnect() throws Exception {
+        EventLoopGroup group = new DefaultEventLoopGroup(1);
+        final LocalAddress addr = new LocalAddress("a");
+        final HttpHeaders headers = new DefaultHttpHeaders();
+        final DefaultFullHttpResponse response = new DefaultFullHttpResponse(
+            HttpVersion.HTTP_1_1,
+            HttpResponseStatus.BAD_GATEWAY);
+        response.headers().add("name", "value");
+        response.headers().add(HttpHeaderNames.CONTENT_LENGTH, "0");
+        final AtomicReference<Throwable> exception = new AtomicReference<Throwable>();
+        ChannelFuture sf =
+            new ServerBootstrap().channel(LocalServerChannel.class).group(group).childHandler(
+                new ChannelInitializer<Channel>() {
+
+                    @Override
+                    protected void initChannel(Channel ch) {
+                        ch.pipeline().addFirst(null, new HttpResponseEncoder());
+                        ch.writeAndFlush(response);
+                    }
+                }).bind(addr);
+        new Bootstrap().channel(LocalChannel.class).group(group).handler(
+            new ChannelInitializer<Channel>() {
+                @Override
+                protected void initChannel(Channel ch) {
+                    ch.pipeline().addFirst(null, new HttpProxyHandler(addr));
+                    ch.pipeline().addLast(new ChannelHandlerAdapter() {
```

Comment:
use a `ChannelInboundHandlerAdapter` as `exceptionCaught(....)` is deprecated in `ChannelHandler`